### DISCUSSION TITLE
new tflite for binary recognition

### DISCUSSION THREAD

**Discussion by renderender** [jomjol/AI-on-the-edge-device#3405] [General]
Hi 
i'm trying to train a tflite model for binary recognition for the digital ROIs - here is why:

i have a LED-matrix on my heating-system and i need to know if certain ROIs of the AiOnTheEdge are white or black - basically if the led is on or not. But i get stuck! Everytime i try a different tflite model on my esp32cam - every single model, even the smallest, results in a freezing AIonTheEdge detection.

So i'm wondering if there is any special needs for the tflite model to be? Is there any documentation on how the models must be trained?

**Comment by SybexX** [comment]
https://github.com/jomjol/neural-network-digital-counter-readout
https://github.com/jomjol/neural-network-analog-needle-readout

**Comment by renderender** [comment]
HI SybexX,

thnk you for your advice. I've read a lot, but i cant find the dimensions to which the aiontheedge is resizing the ROI-Data before input into the original tflite - or easier spoken, which dimension should any tflite be trained for usage in AIONTHEEDGE?
Any suggestions?

**Reply by SybexX** [reply]
Have you looked into the https://github.com/jomjol/neural-network-digital-counter-readout/blob/master/Train_CNN_Digital-Readout-Small-v3.ipynb or in others .ipynb? There's actually enough information in there.

![Unbenannt1](https://github.com/user-attachments/assets/4d7397a7-aab1-499c-9459-b4ef8bcc3827)

![Unbenannt](https://github.com/user-attachments/assets/e83850f6-bddd-486d-b14a-5d97007988d6)

**Reply by SybexX** [reply]
You also have to keep in mind that “TensorFlow Lite Micro” does not contain all the quantization features of “TensorFlow”.
https://github.com/jomjol/AI-on-the-edge-device/issues/3269#issuecomment-2405754709

**Comment by renderender** [comment]
Thank you very much. I'm sure this will help me a lot.

**Comment by gebauer** [comment]
Hi @renderender,

I currently also have plans to train a binary model to recognise a small little icon ("hot_water") on the display on my heater.
Have you succeded in training your model? In principle, I would just need to replace the training set 😉😎

Best
Jan

**Comment by jomjol** [comment]
Just be aware, that the software only supports two different tflite-models at a time:
1. digital counter
2. analog counter
If you want to use one of them for a feature recognition, it will translate this into a number and you need to handle this on the server to separate this information.

**Comment by renderender** [comment]
Hi all,

first of all, thank u so much!

i just had a hard time, trying to train a"binary" model for luminance regognition (Black/white) so i went on and tried to train a model with the above linked code (thanx SybexX) but the idea of manipulating the tflite model to use just 4 categorys instead of 11 didn't bring me to success... So i just did in a dirty way and trained a new model with the existing code an 11 categorys and managed the model not to train 7 of 11 categorys. So i got my tflite trained for my 4 posibilities of my heating system...
I don't have any other analog or digital ROIs to recognize - just my single ROI for my LED-display . So i don't have any problems with the resulting number.


What i did, was:


1. i took pictures with my esp32cam via livestream function within the AiOnTheEdge. 
For each a my category i took one picture
2. i resized the picture to the needed width and height (20x32) and saved it as *.jpg (not *.jpeg!)
3. i created syntetic training material (so that i had more than one pic for every category) 
    actually i don't know how to insert code - otherwise i would have attached this part of code...
4. i trained my model with the original train code and got my working tflite...


i hope i didn't miss anything...  If i can help in any way... i'm here
